src/include/duckdb/main/attached_database.hpp

```diff
@@ -74,6 +79,7 @@ class AttachedDatabase : public CatalogEntry {
 	unique_ptr<TransactionManager> transaction_manager;
 	AttachedDatabaseType type;
 	optional_ptr<Catalog> parent_catalog;
+	optional_ptr<StorageExtension> storage_extension;
```

Comment:
Only minor comment looking at this could be having this a:
```suggestion
	const optional_ptr<StorageExtension> storage_extension;
```
so to have that it can't be changed by error by function taking an AttachedDatabase&.

Reply:
fair. Let me see if this works.

Reply:
so clang complains (because one can change the content of the pointer, which we want). Thinking about it more, I think this is ok as the optional ptr that's returned is a copy of the local one, so one can't change the local one. So we can leave it as is and not have a clang warning.